packages/uma-auth-client/src/hooks/useNwcRequester.tsx

```diff
@@ -1,16 +1,38 @@
 import { useEffect, useState } from "react";
+import * as Nip47 from "src/Nip47Types";
 import { NwcRequester } from "src/NwcRequester";
 import { useOAuth } from "./useOAuth";
 
+class TokenRefreshNwcRequester extends NwcRequester {
+  constructor(
+    nwcConnectionUri: string,
+    private tokenRefresh: () => Promise<void>,
+  ) {
+    super(nwcConnectionUri);
+  }
+
+  async executeNip47Request<T>(
+    nip47Method: Nip47.Method,
+    params: unknown,
+    resultValidator: (result: T) => boolean,
+  ): Promise<T> {
+    await this.tokenRefresh();
```

Comment:
Oh interesting, so the thinking is that because this is reentrant, we can just always make this call before sending any request? That works!

Reply:
hm thinking about this more and i think we'd need to handle this differently. It'll reinstantiate the NwcRequester with the updated nwcConnectionUri after this but will need to make the request again.. so maybe this should retrieve the updated nwcConnectionUri and then executeNip47Request always takes in a nwcConnectionUri and calls relayInit again so that the request can execute? Otherwise the app would have to manually make the request again assuming it doesn't get called on render

Reply:
Updated to do the above; what do you think?

Reply:
LGTM!